File: src/app/models/Monster.ts
export class Monster {
    private _id!: number;
    private _name!: string;
    private _description!: string;
    private _price!: number;
    private _category!: string;
    private _stock!: number;
    private _urlImage!: string;
    
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    public get price(): number {
        return this._price;
    }
    public set price(value: number) {
        this._price = value;
    }
    public get category(): string {
        return this._category;
    }
    public set category(value: string) {
        this._category = value;
    }
    public get stock(): number {
        return this._stock;
    }
    public set stock(value: number) {
        this._stock = value;
    }
    public get urlImage(): string {
        return this._urlImage;
    }
    public set urlImage(value: string) {
        this._urlImage = value;
    }
}

File: src/main/java/io/github/jaoxavier/monstershop/domain/entity/Monster.java
package io.github.jaoxavier.monstershop.domain.entity;

import io.github.jaoxavier.monstershop.domain.enums.Category;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Monster {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    private String name;
    private Double price;
    private String category;
    private Integer stock;
    private String urlImage;

    public Monster(String name, Double price, Category category, Integer stock, String urlImage) {
        this.name = name;
        this.price = price;
        this.category = category.toString();
        this.stock = stock;
        this.urlImage = urlImage;
    }
}
